grammars/basic.cson

```diff
@@ -26,7 +26,7 @@ patterns: [
   }
   {
     # comment
-    match: '(?i:REM.*)'
+    match: '^(?i:REM.*|\'.*)'
```

Comment:
Most BASIC dialects support `:` as a statement separator. Remarks are sometimes written as 
```basic
10 PRINT "HELLO WORLD" : REM SAY HELLO 
```

Reply:
Thanks. Updated the RegEx to match midline `: REM` and single-quote comments. PTAL!

![image](https://user-images.githubusercontent.com/9647195/216429813-b66e7be1-7791-4f40-8a08-9111ea44fb7a.png)

Reply:
If the regex is anchored to the start of the string, then matching will fail in the presence of line numbers, such as in
```basic
10 REM SAY HELLO
20 PRINT "HELLO WORLD"
```
Would simply adding word boundaries around `REM` in the regex (i.e. `(?i:\bREM\b.*|\'.*)`) solve the issue?

Reply:
That makes sense. I've updated the regex again to include the word boundaries for `REM` as you suggest. I've also specified that single-quote comments should be highlighted when they occur either at the beginning of the line or when preceded by a space. 

I think this will capture the _majority_ of uses of single quotes for comments without erroneously catching them in the middle of a string. I tested all of the strings below in a regex tester and for validity in QuickBASIC 4.5. Hopefully this should help improve highlighting for those of us primarily using single quotes for comments without hurting the `REM` highlighting in any way.

<img width="556" alt="Screen Shot 2023-02-08 at 3 53 28 PM" src="https://user-images.githubusercontent.com/9647195/217660083-4911ff68-ac58-40ad-8d41-b624ffdb8f67.png">